### ISSUE TITLE
Is there a way to get the last layer of the model as output?

### ISSUE THREAD

**Issue by maricatovictor** [google/deepvariant#371]
My goal is to use DeepVariant in a transfer learning application.

Is it possible to get the last layer as an embedding output?

If yes:
  Is it possible to use it as a Python module to get these embeddings?

Also, is it possible to run DeepVariant in with a VCF input?

Thank you.

**Comment by AndrewCarroll** [comment]
Hi @maricatovictor 

1. Accessing the pre-logit layer (and other layers) is demonstrated in the code here: https://github.com/google/deepvariant/blob/r1.0/deepvariant/modeling.py#L1161

This is probably the best place to start experimenting if you would like to take information from within the layers for other purposes.

2. There is a new (and somewhat experimental) method to force-call on positions in a VCF. I am attaching a PDF with those instructions. Note that this feature is new, and we may not have enough bandwidth to provide full support for issues that arise in development.

This might be what you mean when asking about VCF input. If you are asking whether it is possible to read in other data from FORMAT or INFO field values of a VCF, this is not yet possible, and definite plans for it are not currently on the roadmap.

[(2020-09-28) Tutorial_ Force calling with DeepVariant.pdf](https://github.com/google/deepvariant/files/5440613/2020-09-28.Tutorial_.Force.calling.with.DeepVariant.pdf)

**Comment by maricatovictor** [comment]
@AndrewCarroll 2 was exactly what I meant, thank you!

About accessing pre-logit layer, I understand that is how you do under the hood, yet, is there any user interface through CLI or Python Module that I could use.

As far as I know, in order to use this `endpoint['PreLogits']` approach, I would need to fork the DeepVariant and change the output, am I missing something?

**Comment by AndrewCarroll** [comment]
Hi @maricatovictor 

At this time, we have not put effort into abstracting these components into a module. We'll take your request into account for future development. It is definitely something that we would like to do, but we likely have a few more pressing issues to address.

Yes, right now you would need to fork the code to be able to work with the pre-logits.

Thank you,
Andrew

**Comment by maricatovictor** [comment]
Ok! No problem, thank you for your assistance! :smile:
